src/modules.js

```diff
@@ -0,0 +1,450 @@
+            dataModel.allTasks.customProjects.push(project);
+            //localStorage.removeItem('allTasks');
+            localStorage.setItem('allTasks', JSON.stringify(dataModel.allTasks));
+        };
+    };
+
+    const createAddToProject = (img) => {
+        img.onclick = (e) => {
+            let list = JSON.parse(localStorage.getItem('allTasks'));
+            list = list.customProjects
+            if (list.length == 0) {
+                let div = document.createElement('div');
+                div.classList.add('projectAddPopup');
+                div.innerText= `Create a Project first
+                
+                `;
+                img.parentElement.appendChild(div);
+                let buttonC = document.createElement('button');
+                buttonC.classList.add('.cancelTaskToProjectBtn');
+                buttonC.innerText="Cancel";
+                div.appendChild(buttonC);
+                cancelTaskToProjectHandler(buttonC);
+            } else {
+                let div = document.createElement('div');
+                div.classList.add('projectAddPopup');
+                div.innerText=`Select a project to add this task to:
+
+                `;
+                img.parentElement.appendChild(div);
+                let sel = document.createElement("select")
+                for (let i=0; i< list.length; i++) { 
+                    let opt = document.createElement("option");
+                    opt.value = list[i].name; 
+                    opt.text = list[i].name; 
+                    sel.add(opt)
+                }
+                div.appendChild(sel);
+                let buttonS = document.createElement('button');
+                buttonS.classList.add('.addTaskToProjectBtn');
+                buttonS.innerText="Submit";
+                div.appendChild(buttonS);
+                saveTaskToProject(img, buttonS, sel);
+                let button = document.createElement('button');
+                button.classList.add('.cancelTaskToProjectBtn');
+                button.innerText="Cancel";
+                div.appendChild(button);
+                cancelTaskToProjectHandler(button);
+                return sel
+            }
+        }
+    };
+
+    const pageLoad = () => {
+        let taskStorage = JSON.parse(localStorage.getItem('allTasks'));
+        if (taskStorage == null) {
+            return
+        } else {
+            getTasks(taskStorage);
+            getProjects(taskStorage);
+            localStorage.removeItem('allTasks');
+            localStorage.setItem('allTasks', JSON.stringify(dataModel.allTasks));
+        };
+    };
+
+    const getTasks = (taskStorage) => {
+        taskStorage = taskStorage.tasks
+        if (taskStorage == null) {
+            return
+        } else {
+            for (let i = 0; i< taskStorage.length; i++) {
+                dataModel.allTasks.tasks.push(taskStorage[i]) 
+            };
+            DOM.displayDefaultList();
+        };
+    };
```

Comment:
This is a nice and small function for me to show a few of the design concepts I've mentioned in other comments:
```
        const newTaskList = taskStorage.tasks;
        if (newTaskList === null) {
            return;
        }
        for (const task of newTaskList) {
            dataModel.allTasks.tasks.push(task); 
        }
        DOM.displayDefaultList();
```
So the new variable is const by default and has a semantic name (explaining what we want from it), the guard-clause is here to exit early preventing nesting, triple equals just by default to avoid possibility of type-coercion, am the use of the "syntactic sugar" for-of loop to abstract away the unneeded index.

All things to either prevent problems before they happen. :)